events/dbpersist.go

```diff
@@ -255,3 +255,21 @@ func (p *DbPersistence) readCarSlice(ctx context.Context, rer *RepoEventRecord)
 
 	return buf.Bytes(), nil
 }
+
+func (p *DbPersistence) TakedownRepo(ctx context.Context, usr util.Uid) error {
+	return p.db.Transaction(func(tx *gorm.DB) error {
+		var event RepoEventRecord
+		if err := p.db.Where("repo = ?", usr).Find(&event).Error; err != nil {
+			return fmt.Errorf("failed to find repo event record: %w", err)
+		}
+
+		if err := p.db.Where("repo_event_record_id = ?", event.Seq).Delete(&RepoOpRecord{}).Error; err != nil {
+			return fmt.Errorf("failed to delete repo op records: %w", err)
+		}
+
+		if err := p.db.Delete(&RepoEventRecord{}, "seq = ?", event.Seq).Error; err != nil {
+			return err
+		}
```

Comment:
I'm a little nervous about a database freeze or deadlock coming out of this. We just had an issue with account deletions causing 10+ minute hangs in PDS prod.

One pattern I have seen to avoid that in the past for this kind of thing is to delete up to, eg, 200 rows per transaction in a loop, starting the next scan from the max seq of the last-deleted batch.

I guess we could put this off until after sandbox mode, and repo takedowns are going to be rare, but it feels like leaving a known footgun lying around. Prod BGS is pretty big from the get-go.

Reply:
Yeah, lets make an issue for this. I think that deleting on a simple condition that is easily looked up by an index should be fine, especially for an admin action that happens *very* rarely.